packages/react-query/src/BlockAuthors.tsx

```diff
@@ -48,10 +53,19 @@ function BlockAuthorsBase ({ children }: Props): React.ReactElement<Props> {
       }).catch(console.error);
 
       // subscribe to new headers
-      api.derive.chain.subscribeNewHeads((lastHeader): void => {
+      api.derive.chain.subscribeNewHeads(async (lastHeader: HeaderExtendedWithMapping): Promise<void> => {
         if (lastHeader?.number) {
           const blockNumber = lastHeader.number.unwrap();
-          const thisBlockAuthor = lastHeader.author?.toString();
+          let thisBlockAuthor = ''; //= lastHeader.author?.toString();
+
+          if (lastHeader.author) {
+            thisBlockAuthor = lastHeader.author.toString();
+          } else if (lastHeader.digest.logs && lastHeader.digest.logs[0] && lastHeader.digest.logs[0].asConsensus[1]) {
```

Comment:
`lastHeader.digest.logs[0].asConsensus[1]` this may fail, there is no `isConsensus` check.

Additionally would do a check for `isFunction(api.query.authorMapping?.mapping)` first (probably not in the subscribe itself, can set a flag outside) - only use this branch in those cases.